server/src/main/java/org/elasticsearch/cluster/BatchedSimpleTaskExecutor.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster;
+
+import org.elasticsearch.core.Tuple;
+
+/**
+ * A {@link BatchedTaskExecutor} for tasks that do not return any value as a result of their execution, other than the next cluster state.
+ */
+public abstract class BatchedSimpleTaskExecutor<Task extends ClusterStateTaskListener> extends BatchedTaskExecutor<Task, Void> {
```

Comment:
Instead of a separate Executor, I tried first to use some interface for the tasks to avoid providing `Void` and `null` for tasks w/o any results, but that ended up a bit convoluted. This seemed to be a better approach, with the only downside that the `executeTask` name cannot be reused, so slightly different names, which I think doesn't matter since they are different executors!

Reply:
I am very open for changing the names of this class and its parent class, if you can come up with a nicer one.